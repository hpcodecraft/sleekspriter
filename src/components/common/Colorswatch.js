import React from "react";
import PropTypes from "prop-types";
import classnames from "classnames";

class Colorswatch extends React.Component {
  constructor(props) {
    super(props);
    this.handleClick = this.handleClick.bind(this);
  }

  render() {
    return (
      <div
        className={classnames({
          colorswatch: true,
          selected: this.props.selected,
        })}
        style={{ background: this.props.color }}
        title={this.props.color}
        onClick={this.handleClick}
      />
    );
  }

  handleClick() {
    this.props.action(this.props.color);
  }
}

Colorswatch.propTypes = {
  selected: PropTypes.bool,
  color: PropTypes.string.isRequired,
  action: PropTypes.func.isRequired,
};

export default Colorswatch;
