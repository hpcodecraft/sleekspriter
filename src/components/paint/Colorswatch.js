import React from "react";

class Colorswatch extends React.Component {
  render() {
    return (
      <div
        className="colorswatch"
        style={{background: this.props.color}}
        title={this.props.color}
        onClick={::this.handleClick} />
    );
  }

  handleClick() {
    this.props.action(this.props.color);
  }
}

export default Colorswatch;
