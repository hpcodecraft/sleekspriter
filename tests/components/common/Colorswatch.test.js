import Colorswatch from "components/common/Colorswatch";

describe("Colorswatch", () => {
  const props = {
    color: "#ff0000",
    action: jest.fn(),
  };

  it("should render correctly", () => {
    const wrapper = render(<Colorswatch {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should render with a background color", () => {
    const wrapper = render(<Colorswatch {...props} />);
    expect(wrapper).toMatchSnapshot();
  });

  it("should call an action handler with its color on click", () => {
    const wrapper = shallow(<Colorswatch {...props} />);
    wrapper.find(".colorswatch").simulate("click");
    expect(props.action).toBeCalledWith(props.color);
  });
});
