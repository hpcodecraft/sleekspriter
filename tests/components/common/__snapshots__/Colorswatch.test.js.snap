// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Colorswatch should render correctly 1`] = `
<div
  class="colorswatch"
  style="background:#ff0000"
  title="#ff0000"
/>
`;

exports[`Colorswatch should render with a background color 1`] = `
<div
  class="colorswatch"
  style="background:#ff0000"
  title="#ff0000"
/>
`;
